storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -669,4 +705,8 @@ public static String remoteTransactionIndexFileName(RemoteLogSegmentMetadata rem
         return generateFileNamePrefixForIndex(remoteLogSegmentMetadata) + LogFileUtils.TXN_INDEX_FILE_SUFFIX;
     }
 
+    public static int estimatedEntrySize(Entry entry) {
+        return entry.offsetIndex.sizeInBytes() + entry.timeIndex.sizeInBytes() + (int) entry.txnIndex.file().length();
```

Comment:
I should have been more clear here. Please keep `entry.offsetIndex.sizeInBytes() + entry.timeIndex.sizeInBytes()` since this is the most efficient way to calculate size since it's in-memory computation. Please use Files.size() instead of `(int) entry.txnIndex.file().length();` since txnIndex does not have a `sizeInBytes` method similar to other two.